src/dsl.jl

```diff
@@ -110,159 +50,186 @@ end
-# if we make the @reaction_network macro call the @network_component macro.
+# if we make the @reaction_network macro call the @network_component macro. Instead, these uses the
+# same input, but passes `complete = false` to `make_rs_expr`.
 """
     @network_component
 
 Equivalent to `@reaction_network` except the generated `ReactionSystem` is not marked as
 complete.
 """
-macro network_component(name::Symbol, ex::Expr)
-    make_reaction_system(striplines(ex); name = :($(QuoteNode(name))))
+macro network_component(name::Symbol, network_expr::Expr)
+    make_rs_expr(QuoteNode(name), network_expr; complete = false)
 end
 
-# Allows @network_component $name begin ... to interpolate variables storing a name.
-macro network_component(name::Expr, ex::Expr)
-    make_reaction_system(striplines(ex); name = :($(esc(name.args[1]))))
+# The case where the name contains an interpolation.
+macro network_component(name::Expr, network_expr::Expr)
+    make_rs_expr(esc(name.args[1]), network_expr; complete = false)
 end
 
-macro network_component(ex::Expr)
-    ex = striplines(ex)
+# The case where nothing, or only a name, is provided.
+macro network_component(name::Symbol = gensym(:ReactionSystem))
+    make_rs_expr(QuoteNode(name); complete = false)
+end
 
-    # no name but equations: @network_component begin ... end ...
-    if ex.head == :block
-        make_reaction_system(ex)
-    else  # empty but has interpolated name: @network_component $name
-        networkname = :($(esc(ex.args[1])))
-        return Expr(:block, :(@parameters t),
-            :(ReactionSystem(Reaction[], t, [], []; name = $networkname)))
-    end
+# Handles two disjoint cases.
+macro network_component(expr::Expr)
+    # Case 1: The input is a name with interpolation.
+    (expr.head != :block) && return make_rs_expr(esc(expr.args[1]); complete = false)
+    # Case 2: The input is a reaction network (and no name is provided).
+    return make_rs_expr(:(gensym(:ReactionSystem)), expr; complete = false)
 end
 
-# Returns a empty network (with, or without, a declared name).
-macro network_component(name::Symbol = gensym(:ReactionSystem))
-    return Expr(:block, :(@parameters t),
-        :(ReactionSystem(Reaction[], t, [], []; name = $(QuoteNode(name)))))
+### DSL Macros Helper Functions ###
+
+# For when no reaction network has been designated. Generates an empty network.
+function make_rs_expr(name; complete = true)
+    rs_expr = :(ReactionSystem(Reaction[], t, [], []; name = $name))
+    complete && (rs_expr = :(complete($rs_expr)))
+    return Expr(:block, :(@parameters t), rs_expr)
+end
+
+# When both a name and a network expression is generated, dispatches these to the internal
+# `make_reaction_system` function.
+function make_rs_expr(name, network_expr; complete = true)
+    rs_expr = make_reaction_system(striplines(network_expr), name)
+    complete && (rs_expr = :(complete($rs_expr)))
+    return rs_expr
 end
```

Comment:
I tried to rewrite the `@reaction_network` and `@network_component` macros so that we don't provide repeat code in the two macros.

Reply:
`make_rs_expr` now does all the work (with the `complete` argument being `true` or `false`).

Reply:
There are basically four cases of `@reaction_network` (or `@network_component`) inputs, we have one macro command handling each one separately.